language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/RecordUtil.java

```diff
@@ -50,45 +51,59 @@ public class RecordUtil {
     /**
      * Get completion items list for struct fields.
      *
-     * @param context Language server operation context
-     * @param fields  Map of field descriptors
-     * @param wrapper  Pair of Raw TypeSymbol and broader TypeSymbol
+     * @param context   Language server operation context
+     * @param fieldsMap Map of identifier to record fields.
      * @return {@link List} List of completion items for the struct fields
      */
     public static List<LSCompletionItem> getRecordFieldCompletionItems(BallerinaCompletionContext context,
-                                                                       Map<String, RecordFieldSymbol> fields,
-                                                                       RawTypeSymbolWrapper wrapper) {
+                                                               Map<RecordField.RecordFieldIdentifier,
+                                                               List<RecordField>> fieldsMap) {
         List<LSCompletionItem> completionItems = new ArrayList<>();
-        fields.forEach((name, field) -> {
-            SnippetContext snippetContext = new SnippetContext();
-            String insertText = getRecordFieldCompletionInsertText(field, snippetContext);
+        fieldsMap.forEach((recordFieldIdentifier, fields) -> {
+            completionItems.add(getRecordFieldCompletionItem(context, recordFieldIdentifier, fields));
+        });
+
+        return completionItems;
+    }
 
-            String detail;
-            if (wrapper.getRawType().getName().isPresent()) {
-                detail = NameUtil.getModifiedTypeName(context, wrapper.getRawType()) + "." + name;
+    private static LSCompletionItem getRecordFieldCompletionItem(BallerinaCompletionContext context,
+                                                 RecordField.RecordFieldIdentifier recordFieldIdentifier,
+                                                 List<RecordField> fields) {
+
+        //Assuming the record fields list is not empty at this level
```

Comment:
Let's check if the fields list is empty

Reply:
I think this can't be empty.

Reply:
Yes at this point the completion list shouldn't be empty